src/components/triniti-app-nav/MainNavContent.jsx

```diff
@@ -20,7 +20,30 @@ class MainNavContent extends React.Component {
 
   constructor(props) {
     super(props);
+
+    this.state = {
+      showModal: false,
+      modal: null,
+    };
     this.handleTitleClick = this.handleTitleClick.bind(this);
+    this.handleShowModal = this.handleShowModal.bind(this);
+    this.handleHideModal = this.handleHideModal.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    // @link https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops
+    // if you need to perform a side effect (for example, data fetching or an animation)
+    // in response to a change in props, use componentDidUpdate lifecycle instead.
+
+    // @link https://reactjs.org/docs/react-component.html#componentdidupdate
+    // You may call setState() immediately in componentDidUpdate()
+    // but note that it must be wrapped in a condition like in the example above,
+    // or you’ll cause an infinite loop. It would also cause an extra re-rendering which,
+    // while not visible to the user, can affect the component performance.
+    if (this.props.location.pathname !== prevProps.location.pathname) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ showModal: false });
```

Comment:
when would this need to happen? some kind of async process that updates the route while the modal is open?